cpp/src/NVStrings_convert.cu

```diff
@@ -138,10 +138,17 @@ int NVStrings::stof(float* results, bool todevice)
     thrust::for_each_n(execpol->on(0), thrust::make_counting_iterator<unsigned int>(0), count,
         [d_strings, d_rtn] __device__(unsigned int idx){
             custring_view* dstr = d_strings[idx];
-            if( dstr )
-                d_rtn[idx] = dstr->stof();
-            else
+            if( !dstr )
                 d_rtn[idx] = (float)0;
+            else if( (dstr->compare("NaN",3)==0) )
+                d_rtn[idx] = NAN;
```

Comment:
I can't seem to find where `NAN` or `INFINITY` are defined. 

Are they using  `std::numeric_limits<float>::quiet_NaN()` and `std::numeric_limits<float>::infinity()`.

Since these are all `constexpr`, they should work in device code.

Reply:
These are in <cmath> or <math.h>
Since these are in a kernel, I did not want to risk using a standard library method for this.

Reply:
Tried to use this and caused a compile error:
```
/home/dwendt/proj/custrings/cpp/src/NVStrings_convert.cu(144): error: calling a constexpr __host__ function("quiet_NaN") from a __device__ function("operator()") is not allowed. The experimental flag '--expt-relaxed-constexpr' can be used to allow this.
```

Reply:
There's no more risk involved in using `std::numeric_limits` than using `<cmath>` since they're both host-side definitions.

The benefit with `numeric_limits` functions is that they return _typed_ variables rather than pre-processor text substitution. Furthermore, you get all the benefit of a macro's value being known at compile time since these functions are `constexpr`.

Reply:
It looks like the CUDA Math lib has a function [`nanf`](https://docs.nvidia.com/cuda/cuda-math-api/group__CUDA__MATH__SINGLE.html#group__CUDA__MATH__SINGLE_1g372c640f910303dc4a7f17ce684322c5) to return a quiet NaN in device code, but I can't find an equivalent for `infinity`.

Reply:
> Tried to use this and caused a compile error:
> 
> ```
> /home/dwendt/proj/custrings/cpp/src/NVStrings_convert.cu(144): error: calling a constexpr __host__ function("quiet_NaN") from a __device__ function("operator()") is not allowed. The experimental flag '--expt-relaxed-constexpr' can be used to allow this.
> ```

Yeah, you'd need to add `--expt-relaxed-constexpr` to the compile flags.

Reply:
So I should change the build to avoid using the NAN and INFINITY values from math.h?

Reply:
The `--expt-relaxed-constexpr` flag is used by cudf as well. 
Updated the code to use the std methods for this.